### PR TITLE
feat: support slack based alerts on failure and sla miss

### PR DESCRIPTION
**Pull request by kushsharma**
- Added support for `Failure` hook point

### CODE REVIEW

**Comment by mauliksoneji** [comment]
Please rebase the branch against current main

**Comment by kushsharma** [comment]
> Please rebase the branch against current main

Done @mauliksoneji

File: ext/scheduler/airflow/tests/test_compiled_airflow_template.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def load_file_as_module(filepath):
 class TestCompiledAirflowTemplate(unittest.TestCase):
 
     def test_should_run_compiled_airflow_template(self):
-        compiled_dag_lib = load_file_as_module('../../../../integration_tests/airflow_1/expected_compiled_template.py')
+        compiled_dag_lib = load_file_as_module('../../airflow2/resources/expected_compiled_template.py')
```
[the remark this answers is not in the document]
Comment (kushsharma) [reply ext/scheduler/airflow/tests/test_compiled_airflow_template.py:18]: Sure, will fix, should have caused by goland refactor.

File: Makefile [hunk 2]
```diff
@@ -27,7 +27,7 @@ pack-files:
 generate-proto: ## regenerate protos
 	@echo " > cloning protobuf from odpf/proton"
 	@rm -rf proton/
-	@git -c advice.detachedHead=false clone https://github.com/odpf/proton --depth 1 --quiet --branch main
+	@git -c advice.detachedHead=false clone https://github.com/odpf/proton --depth 1 --quiet --branch optimus-job-event
```
[the remark this answers is not in the document]
Comment (kushsharma) [reply Makefile:30]: Yes, will [merge](https://github.com/odpf/proton/pull/15) and update Makefile

**Pull request merged**